File: Problems/PRIME1 - Prime Generator/solution.java
import java.util.*;
import java.lang.*;

public class Main
{
    static boolean checkPrime(int num){
        int limit = (int) Math.sqrt(num);
        for(int i = 2 ; i <= limit; i++){
            if(num%i == 0){
                return false;
            }
        }
        return true;
    }
    
	public static void main(String[] args) {
        Scanner S = new Scanner(System.in);
        int testcases = S.nextInt();
        int L, R;
        for(int i = 0; i<testcases; i++){
            L = S.nextInt();
            R = S.nextInt();
            
            if(L == 1)
                L = 2;
            
            for(int num = L; num <= R; num++){
                if( num%6 == 1 || num%6 == 5 || num == 2 || num == 3){
                    if(checkPrime(num)){
                        System.out.println(num);
                    }
                }
            }
            //System.out.println();
        }
	}
}

File: Problems/PRIME1 - Prime Generator/solution.py
#same logic as solution.java but not accepted due to TLE.

from math import sqrt

def checkPrime(num):
    for i in range(2, int(sqrt(num)) +1):
        if(num%i == 0):
            return False
    return True
    
n = 1000000000

testcases = int(input())

for _ in range(testcases):
    L, R = map(int,list(str(input()).split()))
    if(L == 1):
        L = 2
    for i in range(L, R+1):
        if( i%6 == 1 or i%6 == 5 or i == 2 or i == 3 ):
            if(checkPrime(i)):
                print(i)
    print()
